### DISCUSSION TITLE
No module named 'onnxruntime' but its installed?

### DISCUSSION THREAD

**Discussion by iwannalikelinux** [comfyanonymous/ComfyUI#6067] [General]
I am very new Linux mint; so far have been able to install StableSwarmUI and generate few pics using cuda on 3060ti

I realized to work with 3d models I should using the workflows tab – I am amazed how intuitive as software StableSwarmUI and  ComfyUI are :)

I installed the package manger to workflow

this is where I encounter problems when i run workflow and node - RemBGSession
i get following error:

RemBGSession+
No module named 'onnxruntime'

My first thought was i need to get onnxruntime

pip install torch==2.0.1+cu118 torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu118
pip install --upgrade onnxruntime-gpu
pip install rembg

I run again same error message?

I have tried ask AI i got this:

pip uninstall torch torchvision torchaudio onnxruntime-gpu -y

# Reinstall PyTorch with CUDA support (make sure to match versions)
pip install torch==2.1.1+cu118 torchvision==0.16.1+cu118 torchaudio==2.1.1+cu118 -f https://download.pytorch.org/whl/torch_stable.html
# Then reinstall ONNX Runtime again.
pip install onnxruntime-gpu==1.14.0

finally i tried:
pip install --upgrade onnxruntime-gpu

same issue 

RemBGSession+
No module named 'onnxruntime'

I really want try different workflows on comfy I have seen…

I cant resolve this issue after 5+ hours I need help, I don’t get python or how I have seemingly installed what was requested yet its not being seen?

Any help would be really appreciated, please remember I was total newbie (I am trying to learn this)

## System Information
- **ComfyUI Version:** v0.3.7-33-g6d1a3f7
- **Arguments:** main.py --port 7821 --extra-model-paths-config //// StableSwarmUI /StableSwarmUI/Data/comfy-auto-model.yaml --preview-method latent2rgb
- **OS:** posix
- **Python Version:** 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0]
- **Embedded Python:** false
- **PyTorch Version:** 2.5.1+cu121
## Devices

- **Name:** cuda:0 NVIDIA GeForce RTX 3060 Ti : cudaMallocAsync
  - **Type:** cuda
  - **VRAM Total:** 8350400512
  - **VRAM Free:** 6787563520
  - **Torch VRAM Total:** 0
  - **Torch VRAM Free:** 0
  - 
/StableSwarmUI/dlbackend/ComfyUI/execution.py", line 328, in execute
22:19:45.445 [Warning] ComfyUI-0 on port 7821 stderr:     output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
22:19:45.445 [Warning] ComfyUI-0 on port 7821 stderr:                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Comment by LukeG89** [comment]
I think you need to install onnxruntime with a different command, since you've installed pytorch with Cuda 11.8: https://onnxruntime.ai/docs/install/#install-onnx-runtime-gpu-cuda-11x

**Comment by iwannalikelinux** [comment]
thanks i tried your link in terminal (after using the venv command) 

when that didn't work just into terminal without venv

(myenv) $ pip install flatbuffers numpy packaging protobuf sympy
pip install onnxruntime-gpu --index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-11/pypi/simple/
Requirement already satisfied: flatbuffers in ./myenv/lib/python3.10/site-packages (24.3.25)
Requirement already satisfied: numpy in ./myenv/lib/python3.10/site-packages (2.2.0)
Requirement already satisfied: packaging in ./myenv/lib/python3.10/site-packages (24.2)
Requirement already satisfied: protobuf in ./myenv/lib/python3.10/site-packages (5.29.1)
Requirement already satisfied: sympy in ./myenv/lib/python3.10/site-packages (1.13.3)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in ./myenv/lib/python3.10/site-packages (from sympy) (1.3.0)
Looking in indexes: https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-11/pypi/simple/
Requirement already satisfied: onnxruntime-gpu in ./myenv/lib/python3.10/site-packages (1.16.2)
Requirement already satisfied: coloredlogs in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (15.0.1)
Requirement already satisfied: flatbuffers in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (24.3.25)
Requirement already satisfied: numpy>=1.21.6 in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (2.2.0)
Requirement already satisfied: packaging in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (24.2)
Requirement already satisfied: protobuf in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (5.29.1)
Requirement already satisfied: sympy in ./myenv/lib/python3.10/site-packages (from onnxruntime-gpu) (1.13.3)
Requirement already satisfied: humanfriendly>=9.1 in ./myenv/lib/python3.10/site-packages (from coloredlogs->onnxruntime-gpu) (10.0)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in ./myenv/lib/python3.10/site-packages (from sympy->onnxruntime-gpu) (1.3.0)

but same issue both times:

RemBGSession+
No module named 'onnxruntime'

thank for the suggestion tho

**Reply by LukeG89** [reply]
I think it has not installed the right version of onnxruntime, you need to use `--force-reinstall`. Try this line:
```
pip install --force-reinstall onnxruntime-gpu --index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-11/pypi/simple/
```

**Reply by iwannalikelinux** [reply]
thanks i migrated version to SwarmUI v0.9.4.0  > that resolved the issue
